src/main/java/com/google/sps/classes/Utility.java

```diff
@@ -32,16 +33,19 @@
   public static final String SQL_LOCAL_USER = "root";
   public static final String SQL_LOCAL_PASSWORD = "";
   
-  // Query to retrieve data from a question. The ? at the end must be replaced in the
-  // prepared statement, can be '1=1' for all questions or a different condition to match
-  // the questions that are needed.
+  // Query to retrieve data from a question. Generates the following table:
+  //
+  // |-----------------Question-----------------|----FollowerCount--------|-----GetUsername-----|------AnswerCount------|
+  // +----+-------+------+----------+-----------+-------------+-----------+----------+----------+-------------+---------+
+  // | id | title | body | asker_id | date_time | question_id | followers | username | asker_id | question_id | answers |
+  // +----+-------+------+----------+-----------+-------------+-----------+----------+----------+-------------+---------+
   public static final String fetchQuestionQuery = "SELECT * FROM Question "
       + "LEFT JOIN (SELECT question_id, COUNT(follower_id) followers FROM QuestionFollower "
-      + "GROUP BY question_id) CountTable ON Question.id=CountTable.question_id "
-      + "LEFT JOIN (SELECT username, id AS asker_id FROM User) NameTable "
-      + "ON Question.asker_id=NameTable.asker_id "
+      + "GROUP BY question_id) FollowerCount ON Question.id=FollowerCount.question_id "
+      + "LEFT JOIN (SELECT username, id AS asker_id FROM User) GetUsername "
+      + "ON Question.asker_id=GetUsername.asker_id "
       + "LEFT JOIN (SELECT question_id, COUNT(id) answers FROM Answer "
-      + "GROUP BY question_id) AnswerTable ON Question.id=AnswerTable.question_id ";
+      + "GROUP BY question_id) AnswerCount ON Question.id=AnswerCount.question_id ";
```

Comment:
I see that the question's ID appears three times in this table. Is it really needed? If not, maybe removing `question_id` from lines 43 and 47 works?

Reply:
but those IDs are needed to perform the joins, i don't know of any other way to do it

Reply:
Ooh, okok, got it. It's all good then!